### PR TITLE
AEA-740: Update the rules for repeat dispensing prescriptions

### PR DESCRIPTION
**Pull request by mfjarvis**

### CODE REVIEW

File: coordinator/src/services/validation/bundle-validator.ts [hunk 1]
```diff
@@ -165,7 +176,7 @@ function verifyMessageIsResource(message: unknown): message is fhir.Resource {
 
 function verifyResourceTypeIsBundle(resource: unknown): resource is fhir.Bundle {
   return verifyMessageIsResource(resource)
-        && resource.resourceType === "Bundle"
+    && resource.resourceType === "Bundle"
```
Comment (harryhmc) [inline coordinator/src/services/validation/bundle-validator.ts:179]: verifying the resourceType is Bundle is presumably the easier check -- could you flip the conditions around the `&&` to make use of short-circuiting?

File: coordinator/src/services/validation/bundle-validator.ts [hunk 2]
```diff
@@ -126,6 +127,15 @@ export function verifyPrescriptionBundle(bundle: fhir.Bundle): Array<errors.Vali
         "https://fhir.nhs.uk/R4/StructureDefinition/Extension-DM-ResponsiblePractitioner",
         "MedicationRequest.extension"
       )
+    ),
+    (medicationRequests: Array<fhir.MedicationRequest>) => verifyValueIdenticalForAllMedicationRequests(
```
Comment (harryhmc) [inline coordinator/src/services/validation/bundle-validator.ts:131]: is it worth adding a test to `request-validator.spec.ts` around this being enforced?

Comment (harryhmc) [reply coordinator/src/services/validation/bundle-validator.ts:131]: also i know you've only added one of the validators here but there has to be a more concise way of creating this array

**Review by harryhmc — APPROVED** [review]

File: coordinator/src/services/validation/bundle-validator.ts (hunk 2 above)
Comment (mfjarvis) [reply coordinator/src/services/validation/bundle-validator.ts:131]: Didn't bother adding a test as we already have tests for the consistency check, and the only difference here is which field is being considered. As for creating the array, Conner's neatened it up on branch AEA-628-improve-request-validator.

**Pull request merged**